test/e2e/auth/certificates.go

```diff
@@ -108,17 +154,247 @@ var _ = SIGDescribe("Certificates API", func() {
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenCSR.UID, createdCSR.UID)
+
+		ginkgo.By("listing")
+		csrs, err := csrClient.List(context.TODO(), metav1.ListOptions{FieldSelector: "spec.signerName=" + signerName})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(len(csrs.Items), 3, "filtered list should have 3 items")
+
+		ginkgo.By("watching")
+		framework.Logf("starting watch")
+		csrWatch, err := csrClient.Watch(context.TODO(), metav1.ListOptions{ResourceVersion: csrs.ResourceVersion, FieldSelector: "metadata.name=" + createdCSR.Name})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("patching")
+		patchedCSR, err := csrClient.Patch(context.TODO(), createdCSR.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(patchedCSR.Annotations["patched"], "true", "patched object should have the applied annotation")
+
+		ginkgo.By("updating")
+		csrToUpdate := patchedCSR.DeepCopy()
+		csrToUpdate.Annotations["updated"] = "true"
+		updatedCSR, err := csrClient.Update(context.TODO(), csrToUpdate, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(updatedCSR.Annotations["updated"], "true", "updated object should have the applied annotation")
+
+		framework.Logf("waiting for watch events with expected annotations")
+		for sawAnnotations := false; !sawAnnotations; {
+			select {
+			case evt, ok := <-csrWatch.ResultChan():
+				framework.ExpectEqual(ok, true, "watch channel should not close")
+				framework.ExpectEqual(evt.Type, watch.Modified)
+				watchedCSR, isCSR := evt.Object.(*certificatesv1beta1.CertificateSigningRequest)
+				framework.ExpectEqual(isCSR, true, fmt.Sprintf("expected CSR, got %T", evt.Object))
+				if watchedCSR.Annotations["patched"] == "true" {
+					framework.Logf("saw patched and updated annotations")
+					sawAnnotations = true
+					csrWatch.Stop()
+				} else {
+					framework.Logf("missing expected annotations, waiting: %#v", watchedCSR.Annotations)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+
+		// /approval subresource operations
+
+		ginkgo.By("getting /approval")
+		gottenApproval, err := f.DynamicClient.Resource(csrResource).Get(context.TODO(), createdCSR.Name, metav1.GetOptions{}, "approval")
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenApproval.GetObjectKind().GroupVersionKind(), certificatesv1beta1.SchemeGroupVersion.WithKind("CertificateSigningRequest"))
+		framework.ExpectEqual(gottenApproval.GetUID(), createdCSR.UID)
+
+		ginkgo.By("patching /approval")
+		patchedApproval, err := csrClient.Patch(context.TODO(), createdCSR.Name, types.MergePatchType,
+			[]byte(`{"metadata":{"annotations":{"patchedapproval":"true"}},"status":{"conditions":[{"type":"ApprovalPatch","status":"True","reason":"e2e"}]}}`),
```

Comment:
Is adding an arbitrary condition via the `/approval` subresource meant to be allowed? I don't see it mentioned anywhere but I could have missed it 👀

Reply:
`May 29 06:20:07.265: FAIL: patched object should have the applied condition, got []v1beta1.CertificateSigningRequestCondition{v1beta1.CertificateSigningRequestCondition{Type:"ApprovalPatch", Reason:"e2e", Message:"", LastUpdateTime:v1.Time{Time:time.Time{wall:0x0, ext:63726330007, loc:(*time.Location)(0x803b680)}}}}` 😅

Reply:
in general, arbitrary conditions are allowed

CSR's treatment of Approved/Denied/Failed is the only special bit... this ensures other conditions behave normally

Reply:
Ah, it seems you can, but the LastTransitionTime may be an issue here?

Reply:
nah, just a typecast on the Type field:
```
Expected
    <v1beta1.RequestConditionType>: ApprovalPatch
to equal
    <string>: ApprovalPatch
```